dist/src/main/java/io/camunda/application/commons/condition/ConditionalOnDatabaseEnabled.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.application.commons.condition;
+
+import static io.camunda.search.connect.configuration.DatabaseConfig.DATABASE_TYPE_CONFIGURATION;
+
+import io.camunda.application.commons.condition.ConditionalOnDatabaseEnabled.OnDatabaseEnabledCondition;
+import io.camunda.search.connect.configuration.DatabaseConfig;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Documented
+@Conditional(OnDatabaseEnabledCondition.class)
+public @interface ConditionalOnDatabaseEnabled {
+
+  class OnDatabaseEnabledCondition implements Condition {
+    @Override
+    public boolean matches(final ConditionContext context, final AnnotatedTypeMetadata metadata) {
+      final Environment env = context.getEnvironment();
+      final var databaseType = env.getProperty(DATABASE_TYPE_CONFIGURATION, String.class);
```

Comment:
❓ would it also make sense to do some validation here? i.e. fail if the value does not match any of the valid database types?

Reply:
Hm... yes and no 😄 Thinking of making the persistence pluggable, I, as a user, might implement my MongoDB persistence and use the Camunda configuration. I don't have a clear picture of what this would look like (i.e., how to configure, etc.), but at the moment, I would tend not to do further validation here.